### DISCUSSION TITLE
Cross-platform?

### DISCUSSION THREAD

**Discussion by katastrophe92** [celzero/rethink-app#1228] [Ideas]
Hello RethinkDNS team and community! Is a similar app planned for windows also? Kind regards!

**Comment by ghost** [comment]
> Hello RethinkDNS team and community! Is a similar app planned for windows also? Kind regards!
https://github.com/celzero/rethink-app/discussions/945#discussioncomment-6391394

**Comment by ignoramous** [comment]
Some are worried that the Android app in itself isn't stable ... #911 :D

We will attempt porting to Android TV, then Linux Desktop, in that order.

**Reply by ghostersk** [reply]
for linux it would be aweseome. Especially because i could use Wireguard with it, without additional client.
Portmaster does not allow vpn, only for the premium package, and I think you can only use the VPN they provide.

**Reply by Onxr** [reply]
> We will attempt porting to Android TV, then Linux Desktop, in that order.

Eagerly waiting for Linux. But I hope it won't be just a .deb file for release like many other projects

**Reply by Onxr** [reply]
I want to add something.
For Linux when you have the public repository available I guess other people will contribute. I can be wrong but AFAIK Linux crowd prefer FOSS as much as they can and a lot of people try to contribute whatever they can.
So if we have a public repository maybe we will have contribution ?!? We might have more contribution for android version too as more people will know about it. This is just my opinion.

Like I said before I'm eagerly waiting for the Linux port and this app became one of those which I can't live without even though it has some issues. I hope for a good future of Rethink.

**Reply by Onxr** [reply]
Release note from v055v

> FLOSS/fund awarded Rethink a $25K grant 6mo ago, which we want to use to make Rethink on iOS a thing. Let's see.

Is Linux release delayed ? BTW I don't have any Android TV. What's the difference compared to regular Android. Won't current Github release be installed on Android TV ?

**Reply by ignoramous** [reply]
>  Won't current Github release be installed on Android TV ?

It can be, but the UI aspect (16:9) and the UI primitives are very different (touch vs D-Pad) to handhelds. Google *mandates* Android TV apps have abide by their [TV guidelines](https://developer.android.com/docs/quality-guidelines/tv-app-quality). Rethink will be released as a standalone app on Google Play Store. We're close. A *lot* of the work on this was contributed by a community member (but a lot remains to be done, as not every functionality has been ported over... though, I imagine, that'll happen gradually over time).

> Is Linux release delayed ? 

We're doing iOS because on Linux, there's OpenSnitch & LittleSnitch (it wasn't announced back when we had planned this). Nothing similar to Rethink on iOS, otherwise?

We're yet to begin work on iOS; we intend to start soon after Android TV is done.

**Reply by Onxr** [reply]
> Rethink will be released as a standalone app on Google Play Store.

I hope we'll still have Github release. As these days I don't use play store anymore only Obtainium (actually an fork called ObtainX) and if required Aurora store.

>  on Linux, there's OpenSnitch & LittleSnitch

I haven't tried yet. But I was looking for it and found OpenSnitch has issues with WireGuard. I like rethink because of its UX and everything is in one place (DNS, Blocklist, Firewall, VPN, Proxy). I would like to have similar experience on Linux.

BTW as mentioned in #2178. I found two projects for Android and Linux desktop with Amnezia support I also mentioned there.

- [WG Tunnel Android](https://github.com/wgtunnel/android)
- [WG Tunnel Desktop](https://github.com/wgtunnel/desktop)

I haven't tried the desktop version but the Android supports proxy.

**Comment by yoshimo** [comment]
There is also https://github.com/safing/portmaster for other platforms

**Comment by iulko** [comment]
> There is also https://github.com/safing/portmaster for other platforms

They dont even offer offline installer and do an annoying module online installer that you cant really verify. Not real open source in my book. 

Rethink is so much more clear in their mission and even did loopback option just to show what their own app is doing. I cant stress enaught how much I would be happy to see Rethink on other platforms like Windows, but I know its super hard to do.

**Reply by ignoramous** [reply]
Thanks for your kind words.

*Loopback* mode is actually the biggest cause of bugs since `v055b`. I must add though, *IPv4/IPv6* related issues are more painful to resolve, especially with WireGuard also in-place.

**Reply by iulko** [reply]
Yeah I feel like there was alot of big changes in last versions so I understand it will take some time to fix all the bugs
